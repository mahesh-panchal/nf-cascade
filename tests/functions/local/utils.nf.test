
nextflow_function {

    name "Test Functions"
    script "functions/local/utils.nf"

	

    test("Test Function readWithDefault - with file") {

        function "readWithDefault"

        when {
            function {
                """
                input[0] = "$projectDir/configs/multiqc.config"
                input[1] = Channel.value([])
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

    test("Test Function readWithDefault - no file") {

        function "readWithDefault"

        when {
            function {
                """
                input[0] = ""
                input[1] = Channel.fromPath("$projectDir/configs/multiqc.config")
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }
	

    test("Test Function resolveFileFromDir") {

        function "resolveFileFromDir"

        when {
            function {
                """
                input[0] = 'multiqc.config'
                input[1] = Channel.fromPath("$projectDir/configs")
                """
            }
        }

        then {
            assert function.success
            assert snapshot(function.result).match()
        }

    }

    // TODO: Update test with success
    test("Test Function createMagSamplesheet - empty Dir") {

        function "createMagSamplesheet"

        when {
            function {
                """
                input[0] = ''
                """
            }
        }

        then {
            assert !function.success
            assert snapshot(function.result).match()
        }

    }
	

}