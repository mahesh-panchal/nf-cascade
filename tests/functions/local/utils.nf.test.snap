{
    "Test Function readWithDefault - with file": {
        "content": [
            {
                
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-15T09:50:59.021485066"
    },
    "Test Function resolveFileFromDir": {
        "content": [
            {
                
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-15T09:51:10.106910178"
    },
    "Test Function readWithDefault - no file": {
        "content": [
            {
                
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-15T09:53:16.773511965"
    }
}